Tests not running inside docker container in 2.0.1 (DevServices kicking in when it should not?)

## Describe the bug
Our test inside docker container (with test DB in a different container) work on 2.0.0.Final they don´t work in 2.0.1.Final

I believe DevServices is jumping because of something: It does not pick up that we are really using a Database from another container. Not sure if it´s because we use profiles or what´s the reason.

I´ve tried to use `quarkus.datasource.devservices.enabled=false` to disable the behaviour, but it didn´t work. Same error hapenning.

### Expected behavior
Tests to run

### Actual behavior
In 2.0.1.Final, they don´t run inside docker.

## To Reproduce
We are running our tests inside docker with
```
version: '2.1'
services:
  dbtest:
    image: "postgres:11.1"
    environment:
      POSTGRES_USER: user
      POSTGRES_PASSWORD: password
    command: ["-c", "max_connections=500"]

  test:
    image: "12345678.dkr.ecr.eu-west-1.amazonaws.com/builders/ubuntu:v1.0.0_java11_enUS"
    environment:
      - GRADLE_USER_HOME=/gradle/cache
    volumes:
      - ./:/app/workdir
      - $HOME/.gradlecache/:/gradle/cache/
    depends_on:
      dbtest:
        condition: service_started
    command: ["./gradlew", "clean", "check", "-Dquarkus.test.profile=testdocker"]
```
and 

### Configuration
we have the following variables configured to point to that DB.
```properties
%testdocker.quarkus.datasource.db-kind=postgresql
%testdocker.quarkus.datasource.jdbc.url=jdbc:postgresql://dbtest/ourdbname?driver=org.postgresql.Driver
```

### Screenshots
```
15:29:36  test_1    | > Task :compileTestJava
15:29:36  test_1    | warning: unknown enum constant Resolution.OPTIONAL
15:29:36  test_1    |   reason: class file for org.osgi.annotation.bundle.Requirement$Resolution not found
15:29:36  test_1    | warning: unknown enum constant Resolution.OPTIONAL
15:29:36  test_1    |   reason: class file for org.osgi.annotation.bundle.Requirement$Resolution not found
15:29:36  test_1    | warning: unknown enum constant Resolution.OPTIONAL
15:29:36  test_1    |   reason: class file for org.osgi.annotation.bundle.Requirement$Resolution not found
15:29:36  test_1    | warning: unknown enum constant Resolution.OPTIONAL
15:29:36  test_1    |   reason: class file for org.osgi.annotation.bundle.Requirement$Resolution not found
15:29:36  test_1    | warning: unknown enum constant Resolution.OPTIONAL
15:29:36  test_1    |   reason: class file for org.osgi.annotation.bundle.Requirement$Resolution not found
15:29:38  test_1    | 5 warnings
15:29:39  test_1    | 
15:29:39  test_1    | > Task :processTestResources
15:29:39  test_1    | > Task :testClasses
15:29:45  test_1    | No Docker binary found or general error: %s
15:29:45  test_1    | 
15:29:45  test_1    | > Task :test
15:29:45  test_1    | 
15:29:45  test_1    | OurTestResourceTest > ourTestName() FAILED
15:29:45  test_1    |     java.lang.RuntimeException at QuarkusTestExtension.java:699
15:29:45  test_1    |         Caused by: java.lang.RuntimeException at AugmentActionImpl.java:414
15:29:45  test_1    |             Caused by: io.quarkus.builder.BuildException at Execution.java:116
15:29:45  test_1    |                 Caused by: java.lang.IllegalStateException at DevServicesDatasourceProcessor.java:232
15:29:45  test_1    | 
15:29:45  test_1    | 88 tests completed, 1 failed, 83 skipped
````

## Environment 

### Output of `java -version`
JDK11

### GraalVM version (if different from Java)
Not used

### Quarkus version or git rev
2.0.1.Final (not happening on 2.0.0Final)

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)
gradlew 6.8.3

--- a/extensions/datasource/deployment/src/main/java/io/quarkus/datasource/deployment/devservices/DevServicesDatasourceProcessor.java
+++ b/extensions/datasource/deployment/src/main/java/io/quarkus/datasource/deployment/devservices/DevServicesDatasourceProcessor.java
@@ -224,6 +224,18 @@ public class DevServicesDatasourceProcessor {
             return null;
         }
 
+        if (!enabled.isPresent()) {
+            for (DevServicesDatasourceConfigurationHandlerBuildItem i : configHandlers) {
+                if (i.getCheckConfiguredFunction().test(dbName)) {
+                    //this database has explicit configuration
+                    //we don't start the devservices
+                    log.debug("Not starting devservices for " + (dbName == null ? "default datasource" : dbName)
+                            + " as it has explicit configuration");
+                    return null;
+                }
+            }
+        }
+
         if (devDbProvider.isDockerRequired() && !isDockerWorking.getAsBoolean()) {
             String message = "Please configure the datasource URL for "
                     + (dbName == null ? "the default datasource" : " datasource '" + dbName + "'")
@@ -236,18 +248,6 @@ public class DevServicesDatasourceProcessor {
                 return null;
             }
 
-        }
-
-        if (!enabled.isPresent()) {
-            for (DevServicesDatasourceConfigurationHandlerBuildItem i : configHandlers) {
-                if (i.getCheckConfiguredFunction().test(dbName)) {
-                    //this database has explicit configuration
-                    //we don't start the devservices
-                    log.debug("Not starting devservices for " + (dbName == null ? "default datasource" : dbName)
-                            + " as it has explicit configuration");
-                    return null;
-                }
-            }
         }
 
         //ok, so we know we need to start one

